server/src/main/java/org/apache/druid/server/security/TLSUtils.java

```diff
@@ -39,28 +44,77 @@ public static SSLContext createSSLContext(
       String trustStoreType,
       String trustStorePath,
       String trustStoreAlgorithm,
-      PasswordProvider trustStorePasswordProvider
+      PasswordProvider trustStorePasswordProvider,
+      String keyStoreType,
+      String keyStorePath,
+      String keyStoreAlgorithm,
+      String certAlias,
+      PasswordProvider keyStorePasswordProvider,
+      PasswordProvider keyManagerFactoryPasswordProvider
```

Comment:
This method's arg list is getting super long. It would be nicer, although not required, to use a builder instead.

Reply:
Added a builder and replaced constructor calls with builder use